commerce-checkout-web/src/main/java/com/liferay/commerce/checkout/web/internal/util/CommerceCheckoutStepHelper.java

```diff
@@ -61,12 +67,47 @@ public boolean isActivePaymentMethodCommerceCheckoutStep(
 				CommerceCheckoutWebKeys.COMMERCE_ORDER);
 
 		if (_commercePaymentEngine.getCommercePaymentMethodGroupRelsCount(
-				commerceOrder.getGroupId()) > 0) {
+				commerceOrder.getGroupId()) < 1) {
 
-			return true;
+			return false;
 		}
 
-		return false;
+		CommerceContext commerceContext =
+			(CommerceContext)httpServletRequest.getAttribute(
+				CommerceWebKeys.COMMERCE_CONTEXT);
+
+		commerceOrder = _commerceOrderLocalService.recalculatePrice(
+			commerceOrder.getCommerceOrderId(), commerceContext);
+
+		if (BigDecimal.ZERO.compareTo(commerceOrder.getTotal()) == 0) {
+			return false;
+		}
+
+		List<CommercePaymentMethod> commercePaymentMethods =
+			_commercePaymentEngine.getEnabledCommercePaymentMethodsForOrder(
+				commerceOrder.getCommerceOrderId());
+
+		if (commercePaymentMethods.isEmpty()) {
+			return false;
+		}
+
+		if (commercePaymentMethods.size() == 1) {
+			CommercePaymentMethod commercePaymentMethod =
+				commercePaymentMethods.get(0);
+
+			commerceOrder.setCommercePaymentMethodKey(
```

Comment:
@luca-pellizzon  can we create a updateCommerceOrderPaymemntMethod in the service?